### PR TITLE
Second stage of realm transactions

### PR DESCRIPTION
**Pull request by janherich**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Follow-up PR on #4089, porting rest of the entities to the `:data-store/tx` approach, so we are consistent and db writes are not confusing for people unfamiliar with the codebase.

### Steps to test:
Regression testing of the whole app with focus on persistence

status: ready

### CODE REVIEW

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/JpexgT for branch PR-4257

<blockquote><div><strong><a href="https://i.diawi.com/JpexgT">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/ga7frr for branch PR-4257

<blockquote><div><strong><a href="https://i.diawi.com/ga7frr">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/data_store/accounts.cljs [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
-
-(re-frame/reg-fx
- :data-store/save-account
- (fn [{:keys [after-update-event] :as account}]
-   (let [account-to-save (dissoc account :after-update-event)]
-     (async/go (async/>! core/realm-queue #(if after-update-event
-                                             (do (data-store/save account-to-save true)
-                                                 (re-frame/dispatch after-update-event))
-                                             (data-store/save account-to-save true)))))))
+   (assoc coeffects :all-accounts (-> @core/base-realm
```
Comment (cammellos) [inline src/status_im/data_store/accounts.cljs:15]: I find it clearer if we just use `as->` without nesting thread macros, when mixing thread-first thread-last, something like:

```
(as-> @core/base-realm $
    (core/get-all $ :account)
    (core/all-clj  $ :account)
    (map #(update % :settings core/deserialize) $)
```

**Review by cammellos — APPROVED** [review]
Looks good to me! 
I haven't followed the logic as much as I take the behaviour is the same as before.

**Comment by janherich** [comment]
@cammellos Yes, it's exactly the same, quite a simple "mechanical" change. Good point with the readability of the nested threading, I will try to improve it.

**Review by rasom — APPROVED** [review]

**Comment by lukaszfryc** [comment]
@janherich 2 issues I've noticed so far:

1. Object type "account" not found in schema when trying to log in to existing account. I have an app with account and upgraded to the PR build.



2. When I create a new account and log out it's not visible in the list of accounts

**Comment by janherich** [comment]
@lukaszfryc Both issues fixed in the newest build (soon to be completed)

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/ar111H for branch PR-4257

<blockquote><div><strong><a href="https://i.diawi.com/ar111H">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/be7x8T for branch PR-4257

<blockquote><div><strong><a href="https://i.diawi.com/be7x8T">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by yenda — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/X1EecZ for branch PR-4257

<blockquote><div><strong><a href="https://i.diawi.com/X1EecZ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/Hziew7 for branch PR-4257

<blockquote><div><strong><a href="https://i.diawi.com/Hziew7">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@janherich looks good. I tested profile, chat management, etc. + update from nightly. All data looks fine.

**Pull request merged**